src/AppBundle/Notification/SendOnArticlePublished.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace AppBundle\Notification;
+
+use AppBundle\Entity\Article;
+use Doctrine\Common\EventSubscriber;
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use Doctrine\ORM\Event\PreUpdateEventArgs;
+use Doctrine\ORM\Events;
+use GuzzleHttp\Exception\RequestException;
+use Psr\Log\LoggerInterface;
+use sngrl\PhpFirebaseCloudMessaging\ClientInterface;
+use sngrl\PhpFirebaseCloudMessaging\Message;
+use sngrl\PhpFirebaseCloudMessaging\Notification;
+use sngrl\PhpFirebaseCloudMessaging\Recipient\Topic;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+class SendOnArticlePublished implements EventSubscriber
+{
+    const TOPIC = 'news';
```

Comment:
Yes, but I use it in [FirebaseController.php](https://github.com/EnMarche/en-marche.fr/pull/1099/files#diff-2586e748739d54a1be4395bd14ba04b8R34).